### DISCUSSION TITLE
Making a loop never ending

### DISCUSSION THREAD

**Discussion by be80be** [micropython/micropython#15530] [Using MicroPython] [answered]
I know this probably crazy but i cant for the life of me figure how to to loop a while loop to keep looping and not stop.

**Comment by andrewleech** [comment]
``` python
while True:
    do_stuff()
    # if not using asyncio, might want time.sleep() or something to allow background tasks to happen 
    await asyncio.sleep_ms(100)
    # in case you do ever want to break out of the loop
    if _done:
       break
```

**Comment by be80be** [comment]
Here what I was doing 
``import max7219
import time
from machine import Pin, SPI
spi = SPI(1, baudrate=10000000, polarity=0, phase=0)
display = max7219.Matrix8x8(spi, Pin(15), 4)
#while True:
display.brightness(0)
display.fill(0)
display.text('I',0,0,1)
display.show()
time.sleep(1.5)
display.fill(0)
display.show()
display.text('No',0,0,1)
display.show()
time.sleep(1.5)

``

**Reply by be80be** [reply]
Every thing i try threw and error 
MPY: soft reboot
Traceback (most recent call last):
  File "<stdin>", line 7
SyntaxError: invalid syntax

**Reply by robert-hh** [reply]
The error flagged at line 7 could be caused back to line 5. Looking at that line, the symbol Matrix8x8 looks weird. The two x characters look different. But that may just be my PC showing them as such.

**Reply by bixb922** [reply]
Please use 3 backticks to format code, so that we can read it clearly.

May he the problem is that the ```while```  has a comment # before it and the following lines are indented?

**Comment by be80be** [comment]
import max7219
import time
from machine import Pin, SPI
spi = SPI(1, baudrate=10000000, polarity=0, phase=0)
display = max7219.Matrix8x8(spi, Pin(15), 4)
while True:
    do_stuff()
display.brightness(0)
display.fill(0)
display.text('I',0,0,1)
display.show()
time.sleep(1.5)
display.fill(0)
display.show()
display.text('No',0,0,1)
display.show()
time.sleep(1.5)
##await time.sleep_ms(100)
if _done:
       break
This didn't  work still problems 
MPY: soft reboot
Traceback (most recent call last):
  File "<stdin>", line 20, in <module>
SyntaxError: 'break'/'continue' outside loop

**Comment by andrewleech** [comment, marked as answer]
With python the indentation of the file is important, it's part of the core syntax.
Things like `while` or `if` all need their following code to be indented to show what's being run, as opposed to languages like C that use `{` and `}` to show these things

**Comment by be80be** [comment]
""
import max7219
import time
from machine import Pin, SPI
spi = SPI(1, baudrate=10000000, polarity=0, phase=0)
display = max7219.Matrix8x8(spi, Pin(15), 4)
while True:
   
    display.brightness(0)
    display.fill(0)
    display.text('I',0,0,1)
    display.show()
    time.sleep(1.5)
    display.fill(0)
    display.show()
    display.text('Got',0,0,1)
    display.show()
    time.sleep(1.5)
    time.sleep_ms(100)
""

Thanks a bunch I got too work

**Comment by be80be** [comment]
Thanks [andrewleech](https://github.com/andrewleech)
[Andrew Leech](https://github.com/andrewleech)
